### PR TITLE
refactor(web): update to typescript, cleanup small bugs + workspace

### PR DESCRIPTION
**Pull request by dan-online**
I've decided to do some housekeeping with this project for no reason other than I was bored, here's my plan:

- [x] Update main parts of codebase to TypeScript
- [x] Cleanup workspace (add biomejs, commit standards etc)
- [x] CI/CD
- [x] Fix small bugs
  - [x] Change Server choice relaunch to redirect
  - [x] Fix redirect issues
- [x] Refactor main code

### CODE REVIEW

**Comment by elendil7** [comment]
Dan codes strikes again 🦬🐥🐗

**Comment by prayag17** [comment]
Don't change the tauri store to localStorage since I do plan on adding default values to the store for settings. 
I don't plan on refactoring the entire codebase to typescript YET since I don't understand it fully. 
I did plan on adding user credentials encryption by using https://github.com/tauri-apps/tauri-plugin-stronghold, you can do it if you want to

Thanks for your contribution.

**Comment by dan-online** [comment]
> Don't change the tauri store to localStorage since I do plan on adding default values to the store for settings. 
> I don't plan on refactoring the entire codebase to typescript YET since I don't understand it fully. 
> I did plan on adding user credentials encryption by using https://github.com/tauri-apps/tauri-plugin-stronghold, you can do it if you want to
> 
> Thanks for your contribution.
> 
> 

I don't plan on completely removing Tauri store, just checking if it's available, and if so using it, so it can run stand alone in-browser otherwise. I do think I'll also checkout changing it to something like Redux? if that's something you're interested in. As for typescript that's completely fine, I've only updated the first few main files and the store which caught a lot of future bugs, I can leave it like that for now if you'd like.

File: src/components/avatar/avatar.jsx
Comment (prayag17) [inline src/components/avatar/avatar.jsx:1]: Remove AvatarSideMenu component as it is no longer needed

**Review by prayag17 — APPROVED** [review]
Thanks for you contribution :).

**Pull request merged**